### ISSUE TITLE
New processing/p5.js features

### ISSUE THREAD

**Issue by Ramblurr** [quil/quil#391]
There are several "new" processing and p5.js features that would be nice to have available in quil:

* `preload()` - cljs only - [p5.js doc](https://p5js.org/reference/#/p5/preload) 
* Sound -  [p5.js doc](https://p5js.org/reference/#/libraries/p5.sound) [processing doc](https://processing.org/reference/libraries/sound/index.html)

**Comment by dgtized** [comment]
I think sound support should probably remain separate as an extension, just as it is for p5js. I think for now our current focus is stabilizing quil so it's up to date with processing and p5js and easier to release, so adding that much additional surface area to test feels a little out of scope for the moment. That said, a quil-sound extension library that covered both the processing and p5js libraries seems like an excellent idea, and would certainly be a welcome addition if you wanted to take a stab at that.

As for preload(), I'll have to think on that a little. From a quick read through, it's not exactly clear to me what additional functionality it's providing other than blocking on async results in setup? I feel like that's something I would reach to core.async to provide externally and not lean on builtin p5js methods, but maybe that's just me? Could you perhaps outline what you would expect that API would look like in CLJS?

**Comment by Ramblurr** [comment]
> I think sound support should probably remain separate as an extension, just as it is for p5js

I like this idea too. Would you accept it as a new project under the github quil namespace/org?

> From a quick read through, it's not exactly clear to me what additional functionality it's providing other than blocking on async results in setup? I feel like that's something I would reach to core.async to provide externally and not lean on builtin p5js methods, but maybe that's just me? 

I think that's a fair judgement. I already implement my own preload type functionality before instantiating the sketch, I just noticed that preload existed and raised it here as a suggestion.

**Comment by dgtized** [comment]
Yep, happy to import a quil-sound library into the quil namespace/org. I believe we can transfer repos later, so maybe take a stab at creating a prototype and once that has some basic functionality we can move it into the org? Would that work for you?

Re the preload functionality, I'm also happy to revisit later, but also trying to minimize the differences between the Processing and p5js libraries for now to ease in testing/release. It was well worth the suggestion, but let's hold off on it for now.

**Comment by Ramblurr** [comment]
Agree with everything here. When I circle back to the sound lib I'll update y'all.
